internal/server/v1/observations/derived_series.go

```diff
@@ -0,0 +1,507 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package observations implements for /v1/derived/observations/series/...
+package observations
```

Comment:
It would be very helpful to spilt this one file into a few files with modularized functions. I found it a bit hard to follow the structs here.

Reply:
Now split into 3 files. I think in a following PR, it could be further refactored to a more general/abstract calculator that can be shared with obs collection computations.